middlewares/accesslog/logger_test.go

```diff
@@ -357,6 +357,17 @@ func TestNewLogHandlerOutputStdout(t *testing.T) {
 			},
 			expectedLog: `TestHost - TestUser [13/Apr/2016:07:14:19 -0700] "POST testpath HTTP/0.0" 123 12 "testReferer" "testUserAgent" 23 "testFrontend" "http://127.0.0.1/testBackend" 1ms`,
 		},
+		{
```

Comment:
If I add this test case 

```go
{
	desc: "Default config",
	config: &types.AccessLog{
		FilePath: "",
		Format:   CommonFormat,
		Filters:  &types.AccessLogFilters{},
	},
	expectedLog: `TestHost - TestUser [13/Apr/2016:07:14:19 -0700] "POST testpath HTTP/0.0" 123 12 "testReferer" "testUserAgent" 23 "testFrontend" "http://127.0.0.1/testBackend" 1ms`,
},
```

The default response has changed

Reply:
That is right. Now when you specify `Filters` but nothing inside it will remove everything. Isn't that the kind of logical behavior? The filters are OR-connected and when you specify multiple (status codes and retry attempts) than the result is more than when you specify only one of those. WDYT?

Reply:
Thanks for your response @marco-jantke 
When we specify nothing in `Filters`, we need to keep all access logs. And when something is specified in `Filters` we need to keep only access logs matching at least one filter

Reply:
As said before I think that's not logical. And "optimizing" for a use-case where someone would specify an empty filters config doesn't make too much sense. However, I implemented it now as you requested in https://github.com/containous/traefik/pull/3042/commits/6ca142b96ee9a998e5a261e23e97a034e1c0209b. PTAL and let me know what you think. I didn't see a nicer option for detecting an empty configuration except validating each config value one by one to check whether it's the zero value. Note that with this approach we have to extend this method each time a new option gets added. I'm open to improvement ideas.

Reply:
Thanks for your correction @marco-jantke 

I am pretty agree with you but I think that it is not the aim of this PR to change the default behavior. To me you have choose the best option to detect an empty configuration.

We should maybe do an other PR to improve this method to avoid to extend it every time we add a new option